include/swift/AST/ASTContext.h

```diff
@@ -1184,12 +1184,14 @@ class ASTContext final {
   /// module is loaded in full.
   bool canImportModuleImpl(ImportPath::Module ModulePath, SourceLoc loc,
                            llvm::VersionTuple version, bool underlyingVersion,
-                           bool updateFailingList,
-                           llvm::VersionTuple &foundVersion) const;
+                           bool isSourceCanImport,
+                           llvm::VersionTuple &foundVersion,
+                           llvm::VersionTuple &foundUnderlyingClangVersion) const;
```

Comment:
I feel like if we just sink `addSucceededCanImportModule` into `canImportModuleImpl`, there is no need to pass the parameter here. It made the code a bit cleaner that way.

Reply:
I guess I did found we need two versions of the can import check with only one of them updating the mapping so the alternative is passing a flag. Not that much difference so it might be fine.

Reply:
Yeah I share the sentiment, but it's a bit more trouble than it is worth, I think, since there's some value in having this function be `const`. And on top of that, there are multiple `return true` code-paths here and if we ever added new ones, we'd need to ensure that we call `addSucceededCanImportModule` on it before existing, which is trickier to enforce.